homeassistant/components/homeworks/__init__.py

```diff
@@ -87,6 +98,66 @@ class HomeworksData:
     keypads: dict[str, HomeworksKeypad]
 
 
+@callback
+def async_setup_services(hass: HomeAssistant) -> None:
+    """Set up services for Lutron Homeworks Series 4 and 8 integration."""
+
+    async def async_call_service(service_call: ServiceCall) -> None:
+        """Call the service."""
+        await async_send_command(hass, service_call.data)
+
+    hass.services.async_register(
+        DOMAIN,
+        "send_command",
+        async_call_service,
+        schema=SERVICE_SEND_COMMAND_SCHEMA,
+    )
+
+
+async def async_send_command(hass: HomeAssistant, data: Mapping[str, Any]) -> None:
+    """Send command to a controller."""
+
+    def get_controller_ids() -> list[str]:
+        """Get homeworks data for the specified controller ID."""
+        return [data.controller_id for data in hass.data[DOMAIN].values()]
+
+    def get_homeworks_data(controller_id: str) -> HomeworksData | None:
+        """Get homeworks data for the specified controller ID."""
+        data: HomeworksData
+        for data in hass.data[DOMAIN].values():
+            if data.controller_id == controller_id:
+                return data
+        return None
+
+    def send_commands(controller: Homeworks, commands: list[str]) -> None:
+        """Send commands to controller."""
+        _LOGGER.debug("Send commands: %s", commands)
+        for command in commands:
+            if command.lower().startswith("delay"):
+                delay = int(command.partition(" ")[2])
+                _LOGGER.debug("Sleeping for %s ms", delay)
+                sleep(delay / 1000)
```

Comment:
We don't allow sleep in the executor thread pool, since it can exhaust the pool. Sleeping needs to be done with `asyncio.sleep`.

Reply:
Fixed in https://github.com/home-assistant/core/pull/114867